src/libflake/call-flake.nix

```diff
@@ -92,7 +95,7 @@ let
           assert builtins.isFunction flake.outputs;
           result
         else
-          sourceInfo;
+          sourceInfo // { inherit sourceInfo outPath; };
```

Comment:
This seems quite significant but has no test coverage.

I think it also deserves a mention in the [release notes](https://nix.dev/manual/nix/2.28/development/contributing#add-a-release-note) that `sourceInfo` is now a valid attribute of non-flake inputs, and presumably `{ flake = false; url = "...?dir=subdir"; }` now works?

Reply:
Yes, I wanted to get some feedback on the implementation before working on tests etc.

I'll squash and add some tests when I get chance 👍

Reply:
I've added some test coverage and a release note. Please let me know if/how I can improve them.